### PR TITLE
Add WeekDaysUnit

### PR DESCRIPTION
**Pull request by 9swampy**
I've needed to add the ability to run just on weekdays (i.e. not weekend days) - educate me if I've missed a current implementation. Tbh I just needed to run every weekday, didn't need the ability to run every Nth weekday - but that's included in the implementation to work with your pattern; hopefully my presumption to e.g. skip Monday if Every(2).WeekDays() is specified on a weekend is compliant?

Hmm, I'm also going to need to run repeatedly during certain times in the day (weekdays only)...

A little late to the looking for a new owner party (@tallesl https://github.com/fluentscheduler/FluentScheduler/issues/59) but I'd offer to help out, merge and publish an updated NuGet package if you're happy to add me (https://www.nuget.org/profiles/9swampy) as an owner of the package on NuGet.

Kind regards,

J

### CODE REVIEW

**Comment by 9swampy** [comment]
Updated to include the ability to run repeatedly between specified hours and also on on WeekDays only.

@jgeurts If you've a moment to review my implementation it's always good to get input from the original dev :) Offer to Nuget stands.

File: FluentScheduler/Extensions/DateTimeExtensions.cs [hunk 1]
```diff
@@ -90,6 +90,45 @@ public static DateTime Next(this DateTime current, DayOfWeek dayOfWeek)
 			return current.AddDays(offsetDays);
 		}
 
+    /// <summary>
+    /// Nexts the Nth weekday.
+    /// </summary>
+    /// <param name="d">The day.</param>
+    /// <param name="n">The number of weekdays to advance.</param>
+    /// <returns>Advanced date.</returns>
+    public static DateTime NextNWeekday(this DateTime d, int n)
```
Comment (jgeurts) [inline FluentScheduler/Extensions/DateTimeExtensions.cs]: Can you make the variable names be a bit more descriptive than single characters?

**Comment by jgeurts** [comment]
All in all, it looks good to me. That said, I don't have a windows box that I can build/test this on, but hopefully @tallesl does. 

Only suggestions I would make would be to standardize where the using statements are and try to make the variable names as descriptive as possible. Thanks for the PR!

**Comment by tallesl** [comment]
Hey there,

I do have a Windows box, I'm reviewing and merging it soon (couple of days probably).

There's one thing or two to keep the repository "fresh" that I might do before publishing a new package version, so I'll ask you a little patience until it's published (won't take long, I promise!).

Things like tabs vs spaces (the project is using tabs while the default Visual Studio setup uses spaces, [you can see by yourself on the pr diff](https://github.com/fluentscheduler/FluentScheduler/pull/60/files)), clean up those [stale branches](https://github.com/fluentscheduler/FluentScheduler/branches/stale), set up [a CI server](http://appveyor.com). Little things :wink:

(one of them you already did by the way, removing that annoying NuGet restore!)

**Comment by 9swampy** [comment]
Sorry on the namespaces, old habits: http://www.stylecop.com/docs/SA1200.html UsingDirectivesMustBePlacedWithinNamespace. 
NextNWeekDay was a cut/paste from my day-job :-p the rest's descriptive enough though? ...but it was whether I fit with some architectural sense I was really looking for which "all in all" covered.

Thx for putting the Repo up in the first place, "if I see farther"...

**Comment by tallesl** [comment]
Merged.

Since I rebased out those two merge commits, you might want to get a fresh fork if you plan to send another pull request on the future (your fork history doesn't match with the current one here).

**Comment by 9swampy** [comment]
Yeah, I see what you've done, pulled everything back up fastforward onto the master. No problem, I rebased what I've worked on since onto that. Before you publish you might like to consider moving the RestrictableUnitExtensions into the root namespace as I've subsequently found having to add FluentScheduler.Extensions all the time is really irritating.

**Pull request closed without merging**